tests/integration/construct_component_slow/testcomponent/index.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright 2016-2021, Pulumi Corporation.  All rights reserved
+
+import * as pulumi from "@pulumi/pulumi";
+import * as dynamic from "@pulumi/pulumi/dynamic";
+import * as provider from "@pulumi/pulumi/provider";
+
+// The component has a child resource that takes a long time to be created.
+// We want to ensure the component's slow child resource will actually be created when the
+// component is created inside `construct`.
+
+const CREATION_DELAY = 15 * 1000; // 15 second delay in milliseconds
```

Comment:
I'm generally not a huge fan of introducing delays like this in tests. I was thinking about having the provider spin until it received some signal, but I couldn't think of a good place where to send the signal in the tests. Open to other ideas.

15 seconds seemed to be the least amount of time to reliably reproduce to issue, at least on my machine. Any less and sometimes the tests would pass without the SDK change.